share/wake/lib/system/sources.wake

```diff
@@ -58,6 +58,67 @@ export def source (file: String): Result Path Error =
         x, Nil = Pass x
         _      = unreachable "a quoted RegExp can never match more than one distinct String"
 
+# Allows claiming of a file so long as the file is within
+# wakeroot. Keep in mind that if another job produces this
+# file, claim will fail. Likewise if you claim a file then
+# any job that produces this file will fail. Claim should
+# be thought of as a "virtual" job and wake demands that
+# all paths be produced by exactly one job, virtual or
+# otherwise.
+#
+# For source files please use `source`. For files outside
+# of wakeroot please use `claimFileAsPath` or `claimFileAsPathIn`.
+# `claim` is good for claiming files of previous builds that
+# are not being used again this build.
+#
+# Parameters:
+#  - `file`: The string giving the path to the file to be claimed
+#
+# Example:
+# ```
+#   claim "build/my_flow/my_step/previous-output.txt" # Claims the file
+# ```
+export def claim (file: String): Result Path Error =
+    # Define the actual virtual job that produces the Path
+    def raw_claim file =
+        def get_modtime file = prim "get_modtime"
+        def time = get_modtime file
+        if time == -1 then
+            failWithError "{file}: files does not exist. Maybe you haven't previously built this file?"
+        else
+            # Its important that the command here is distinct from what `source` would use
+            # so that claim and source do not overlap.
+            makeExecPlan ("<claim>", str time, file, Nil) Nil
+            | setPlanShare       False
+            | setPlanEcho        logDebug
+            | setPlanEnvironment Nil
+            | setPlanFnOutputs   (file, _)
+            | runJobWith virtualRunner
+            | getJobOutput
+
+    # Compute the relative path in simplist form
+    def isAbs = matches `/.*` file
+    def relPath = if isAbs then
+        relative workspace file
+    else
+        simplify file
+
+    # Fail if we're outside of wakeroot (relPath must be simplified)
+    def isOutsideRoot = matches `\.\..*/.*` relPath
+    require False = isOutsideRoot
+    else failWithError "claim cannot be used on files outside of wake's root. Consider using claimFileAsPath instead."
+
+    # Now that our path is uniform check if it could have been fetched with `source`
+    def base = basename relPath
+    def dir = dirname relPath
+    def scan dir regexp = prim "sources"
+    def canBeSourced = exists (_==~relPath) (scan dir base.quote)
```

Comment:
These all just *barely* squeak through as exceptions to (my interpretation of) the newline style.  Really, I'm only all right with `canBeSourced` because it's hiding among everything else in the block which has legitimate reasons for being single lines, but that one in particular should probably be broken up.

Reply:
I left a comment elsewhere about this but I'm willing to expend some social capital, even go in debt, to fight this standard. The current rules are not good for automation, learnability, and are too restrictive in many many cases IMO. Even the strictest standards you see elsewhere don't even come close to suggesting these be reformatted.